packages/react-debug-tools/src/ReactDebugHooks.js

```diff
@@ -509,6 +516,40 @@ function useFormState<S, P>(
   return [state, (payload: P) => {}];
 }
 
+function useHostTransitionStatus(): TransitionStatus {
+  // The type is host specific. This is just a placeholder.
+  let status: mixed = null;
+
+  if (currentFiber !== null) {
+    const dependencies = currentFiber.dependencies;
+    if (dependencies !== null) {
+      let contextDependency = dependencies.firstContext;
+      while (contextDependency !== null) {
+        if (
+          // $FlowFixMe -- TODO
+          contextDependency.context.$$contextType ===
+          Symbol.for('react.HostTransitionContext')
+        ) {
+          // TODO: This as well as contextDependency.context.currentValue is null because setupContext does not set the value yet
+          status = contextDependency.memoizedValue;
```

Comment:
Will rebase this onto https://github.com/facebook/react/pull/28467 and then we just read from `currentContextDependency.memoizedValue`